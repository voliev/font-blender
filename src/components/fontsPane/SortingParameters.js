import React from "react";
import PropTypes from "prop-types";
import "./styles/SortingParameters.css";

const SortingParameters = ({ sortingParamChangeHandler, value }) => (
  <select
    className="SortingParameters"
    id="sorting-param"
    name="param"
    onChange={sortingParamChangeHandler}
    title="Select sorting parameter"
    value={value}
  >
    {["popularity", "alphabetical"].map(param => (
      <option value={param === "alphabetical" ? "alpha" : param} key={param}>
        {param}
      </option>
    ))}
  </select>
);

SortingParameters.propTypes = {
  sortingParamChangeHandler: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired
};

export default SortingParameters;
