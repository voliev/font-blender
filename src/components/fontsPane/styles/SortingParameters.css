.SortingParameters {
  height: 2rem;
  margin-left: var(--space-lg);
  text-transform: capitalize;
  width: 15rem;
}
