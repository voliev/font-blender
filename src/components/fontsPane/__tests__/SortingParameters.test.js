import React from "react";
import { mount, shallow } from "enzyme";
import SortingParameters from "../SortingParameters";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    value: "popularity",
    sortingParamChangeHandler: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<SortingParameters {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { wrapper, value, sortingParamChangeHandler } = setup(mount);

  expect(typeof wrapper.prop("value")).toBe("string");
  expect(wrapper.prop("value")).toBe(value);
  expect(typeof wrapper.prop("sortingParamChangeHandler")).toBe("function");
  expect(wrapper.prop("sortingParamChangeHandler")).toBe(
    sortingParamChangeHandler
  );
});

it("should set corresponding value on initial render", () => {
  const { wrapper } = setup();

  expect(wrapper.prop("value")).toBe("popularity");
});

it("should call handler parameter on value change", () => {
  const { wrapper, sortingParamChangeHandler } = setup();

  const event = {
    target: { value: "alpha" }
  };
  wrapper.find("select").simulate("change", event);

  expect(sortingParamChangeHandler).toHaveBeenCalled();
  expect(sortingParamChangeHandler).toHaveBeenCalledTimes(1);
  expect(sortingParamChangeHandler).toHaveBeenCalledWith(event);
});
