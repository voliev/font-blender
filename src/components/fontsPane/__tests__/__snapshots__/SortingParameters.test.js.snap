// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<select
  className="SortingParameters"
  id="sorting-param"
  name="param"
  onChange={[MockFunction]}
  title="Select sorting parameter"
  value="popularity"
>
  <option
    key="popularity"
    value="popularity"
  >
    popularity
  </option>
  <option
    key="alphabetical"
    value="alpha"
  >
    alphabetical
  </option>
</select>
`;
